### DISCUSSION TITLE
Types migration guide

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [JedWatson/react-select#4822] [Q&A]
Hi. Where i can find a types migration guide from v4 to v5?

I cant find OptionTypeBase analog

**Comment by Rall3n** [comment]
Hello @popuguytheparrot,

there is no analog for `OptionsTypeBase` because there is no need for it anymore.

As stated in the [upgrade guide for `v4 -> v5`](https://react-select.com/upgrade#from-v4-to-v5):

> Options can be any type (if using getOptionValue and getOptionLabel) so there's no longer a base type for options

**Comment by popuguytheparrot** [comment]
I have annotation like this
```js
  onChangeNewProduct: (value: OpportunityApi.OpportunityForCreateEntity, action?: ActionMeta<OptionTypeBase>) => void;
```

**Reply by Methuselah96** [reply]
You can use `unknown` in place of `OptionTypeBase` there.

**Comment by popuguytheparrot** [comment]
<html>TS2769: No overload matches this call.<br/>Overload 2 of 2, '(props: StyledComponentPropsWithAs&lt;AsyncSelect, any, { invalid: boolean; }, never, AsyncSelect, AsyncSelect&gt;): ReactElement&lt;...&gt;', gave the following error.<br/>Type 'MutableRefObject&lt;HTMLSelectElement | undefined&gt;' is not assignable to type 'Ref&lt;Select&lt;unknown, boolean, GroupBase&lt;unknown&gt;&gt;&gt; | undefined'.<br/>Overload 2 of 2, '(props: StyledComponentPropsWithAs&lt;AsyncSelect, any, { invalid: boolean; }, never, AsyncSelect, AsyncSelect&gt;): ReactElement&lt;...&gt;', gave the following error.<br/>Type 'boolean | undefined' is not assignable to type 'boolean'.<br/>Overload 2 of 2, '(props: StyledComponentPropsWithAs&lt;AsyncSelect, any, { invalid: boolean; }, never, AsyncSelect, AsyncSelect&gt;): ReactElement&lt;...&gt;', gave the following error.<br/>Type '(props: ISuggestion) =&gt; JSX.Element' is not assignable to type 'GetOptionLabel&lt;unknown&gt;'.<br/>Overload 2 of 2, '(props: StyledComponentPropsWithAs&lt;AsyncSelect, any, { invalid: boolean; }, never, AsyncSelect, AsyncSelect&gt;): ReactElement&lt;...&gt;', gave the following error.<br/>Type '(option: ISuggestion) =&gt; string' is not assignable to type 'GetOptionValue&lt;unknown&gt;'.<br/>Overload 2 of 2, '(props: StyledComponentPropsWithAs&lt;AsyncSelect, any, { invalid: boolean; }, never, AsyncSelect, AsyncSelect&gt;): ReactElement&lt;...&gt;', gave the following error.<br/>Type '(option: ISuggestion) =&gt; boolean | undefined' is not assignable to type '(option: unknown, selectValue: Options&lt;unknown&gt;) =&gt; boolean'.<br/>Overload 2 of 2, '(props: StyledComponentPropsWithAs&lt;AsyncSelect, any, { invalid: boolean; }, never, AsyncSelect, AsyncSelect&gt;): ReactElement&lt;...&gt;', gave the following error.<br/>Type '{ clearIndicator: (styles: CSSProperties) =&gt; { cursor: string; alignContent?: Property.AlignContent | undefined; alignItems?: Property.AlignItems | undefined; alignSelf?: Property.AlignSelf | undefined; ... 779 more ...; vectorEffect?: Property.VectorEffect | undefined; }; dropdownIndicator: (styles: CSSProperties) ...' is not assignable to type 'StylesConfig&lt;unknown, boolean, GroupBase&lt;unknown&gt;&gt;'.<br/>Overload 2 of 2, '(props: StyledComponentPropsWithAs&lt;AsyncSelect, any, { invalid: boolean; }, never, AsyncSelect, AsyncSelect&gt;): ReactElement&lt;...&gt;', gave the following error.<br/>Type '{ ValueContainer: (props: any) =&gt; JSX.Element; MultiValue: (props: MultiValueProps&lt;any&gt;) =&gt; JSX.Element; SingleValue: (props: any) =&gt; JSX.Element; Menu: (props: MenuProps&lt;any, boolean&gt;) =&gt; JSX.Element; MenuList: (props: MenuListProps&lt;any, boolean&gt;) =&gt; JSX.Element; Option: (props: OptionProps&lt;...&gt;) =&gt; JSX.Element; Op...' is not assignable to type 'Partial&lt;SelectComponents&lt;unknown, boolean, GroupBase&lt;unknown&gt;&gt;&gt;'.<br/>Object literal may only specify known properties, and 'OptionLabel' does not exist in type 'Partial&lt;SelectComponents&lt;unknown, boolean, GroupBase&lt;unknown&gt;&gt;&gt;'.

**Reply by Methuselah96** [reply]
Can you provide a code sample that reproduces this error?
